mteb/tasks/Retrieval/multilingual/WikipediaRetrievalMultilingual.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+from datasets import load_dataset
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks import MultilingualTask
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+
+_EVAL_LANGS = {
+    "bg": ["bul-Cyrl"],
+    "bn": ["ben-Beng"],
+    "cs": ["ces-Latn"],
+    "da": ["dan-Latn"],
+    "de": ["deu-Latn"],
+    "en": ["eng-Latn"],
+    "fa": ["fas-Arab"],
+    "fi": ["fin-Latn"],
+    "hi": ["hin-Deva"],
+    "it": ["ita-Latn"],
+    "nl": ["nld-Latn"],
+    "pt": ["por-Latn"],
+    "ro": ["ron-Latn"],
+    "sr": ["srp-Cyrl"],
+    "no": ["nor-Latn"],
+    "sv": ["swe-Latn"],
+}
+
+
+# adapted from MIRACLRetrieval
+def _load_data(
+    path: str,
+    langs: list,
+    split: str,
+    cache_dir: str = None,
+    revision_queries: str = None,
+    revision_corpus: str = None,
+    revision_qrels: str = None,
+):
+    queries = {lang: {split: {}} for lang in langs}
+    corpus = {lang: {split: {}} for lang in langs}
+    qrels = {lang: {split: {}} for lang in langs}
+
+    for lang in langs:
+        print(f"LOADING {lang}")
+        queries_path = path
+        corpus_path = path.replace("queries", "corpus")
+        qrels_path = path.replace("queries", "qrels")
+        queries_lang = load_dataset(
+            queries_path,
+            lang,
+            split=split,
+            cache_dir=cache_dir,
+            revision=revision_queries,
+        )
+        corpus_lang = load_dataset(
+            corpus_path,
+            lang,
+            split=split,
+            cache_dir=cache_dir,
+            revision=revision_corpus,
+        )
+        qrels_lang = load_dataset(
+            qrels_path,
+            lang,
+            split=split,
+            cache_dir=cache_dir,
+            revision=revision_qrels,
+        )
+        # don't pass on titles to make task harder
+        corpus_lang_dict = {doc["_id"]: {"text": doc["text"]} for doc in corpus_lang}
+        queries_lang_dict = {
+            query["_id"]: {"text": query["text"]} for query in queries_lang
+        }
+        # qrels_lang_dict = {qrel["query-id"]: {qrel["corpus-id"]: qrel["score"]} for qrel in qrels_lang}
+
+        qrels_lang_dict = {}
+        for qrel in qrels_lang:
+            if qrel["score"] == 0.5:
+                continue
```

Comment:
Unfortunately, I couldn't get `WikiepediaRetrievalMultilingual` running with the hard negatives. I assumed one can just set scores inbetween 0 and 1 in the qrels dataset, e.g. 0.5 for hard negatives.
https://huggingface.co/datasets/ellamind/wikipedia-2023-11-retrieval-multilingual-qrels

But MTEB expects `int` at some point so I implemented a workaround that just drops all the hard negatives. Would be nice, if we don't waste them.

Reply:
I would create an issue on this, then we can come back to it later